bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/Car.java

```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    public void setHVACStatus(JsonObject responseJson) {
+        try {
+            JsonObject attributes = getAttributes(responseJson);
+            if (attributes != null && attributes.get("hvacStatus") != null) {
+                hvacstatus = attributes.get("hvacStatus").getAsString().equals("on");
+            }
+        } catch (IllegalStateException | ClassCastException e) {
+            logger.warn("Error {} parsing HVAC Status: {}", e.getMessage(), responseJson);
+        }
+    }
+
+    public void setCockpit(JsonObject responseJson) {
+        try {
+            JsonObject attributes = getAttributes(responseJson);
+            if (attributes != null && attributes.get("totalMileage") != null) {
+                odometer = attributes.get("totalMileage").getAsDouble();
+            }
+        } catch (IllegalStateException | ClassCastException e) {
+            logger.warn("Error {} parsing Cockpit: {}", e.getMessage(), responseJson);
+        }
+    }
+
+    public void setLocation(JsonObject responseJson) {
+        try {
+            JsonObject attributes = getAttributes(responseJson);
+            if (attributes != null) {
+                if (attributes.get("gpsLatitude") != null) {
+                    gpsLatitude = attributes.get("gpsLatitude").getAsDouble();
+                }
+                if (attributes.get("gpsLongitude") != null) {
+                    gpsLongitude = attributes.get("gpsLongitude").getAsDouble();
+                }
+            }
+        } catch (IllegalStateException | ClassCastException e) {
+            logger.warn("Error {} parsing Cockpit: {}", e.getMessage(), responseJson);
+        }
+    }
+
+    public void setDetails(JsonObject responseJson) {
+        try {
+            if (responseJson.get("assets") != null) {
+                JsonArray assetsJson = responseJson.get("assets").getAsJsonArray();
+                for (JsonElement asset : assetsJson) {
+                    if (asset.getAsJsonObject().get("assetType") != null
+                            && asset.getAsJsonObject().get("assetType").getAsString().equals("PICTURE")) {
+                        if (asset.getAsJsonObject().get("renditions") != null) {
+                            JsonArray renditions = asset.getAsJsonObject().get("renditions").getAsJsonArray();
+                            for (JsonElement rendition : renditions) {
+                                if (rendition.getAsJsonObject().get("resolutionType") != null
+                                        && rendition.getAsJsonObject().get("resolutionType").getAsString()
+                                                .equals("ONE_MYRENAULT_SMALL")) {
+                                    imageURL = rendition.getAsJsonObject().get("url").getAsString();
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    if (imageURL.length() > 0) {
```

Comment:
First, it is better to use `!imageURL.isEmpty()`
But the real problem is that imageURL is a class member that could already have a value before this method is called. So you should probably use a local variable first and then finally set imageURL.

Reply:
OK that is the isEmpty() method... Got it.

Reply:
Done